dashboard/modules/log/log_agent.py

```diff
@@ -20,3 +25,101 @@ async def run(self, server):
     @staticmethod
     def is_minimal_module():
         return False
+
+
+BLOCK_SIZE = 8192
+
+
+class LogAgentV1Grpc(
+    dashboard_utils.DashboardAgentModule, reporter_pb2_grpc.ReporterServiceServicer
+):
+    def __init__(self, dashboard_agent):
+        super().__init__(dashboard_agent)
+
+    async def run(self, server):
+        if server:
+            reporter_pb2_grpc.add_LogServiceServicer_to_server(self, server)
+
+    # TODO: should this return True
+    @staticmethod
+    def is_minimal_module():
+        return False
+
+    async def LogIndex(self, request, context):
+        if os.path.exists(self._dashboard_agent.log_dir):
```

Comment:
Hmm this should always exist. Did you see any edge case it doesn't exist?

Reply:
It should always exist, but you mentioned to check it just to avoid the uncaught error